src/NUnitFramework/framework/Internal/Execution/EventQueue.cs

```diff
@@ -15,13 +29,20 @@ namespace NUnit.Framework.Internal.Execution
     /// or to queue them for forwarding on another thread or at
     /// a later time.
     /// </summary>
-    public abstract class Event
+    public abstract class Event : IEvent<ITestListener>, IDisposable
```

Comment:
Should an `Event` not be a pure data class?
Who would `Dispose` an event. Normally code creating would do that, but they don't know when it is received. A receiver can't do it as we don't know how many listeners there are.

Reply:
Yes, that was the intention, but the compiler complained with the declaration for EventQueue, and I just dropped in the IDisposable to fix it.  It should not be there, but I didn't find why it complained.